src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2296,15 +2296,22 @@ namespace nupic {
 
   void SpatialPoolerTest::testInitPermConnected()
   {
+    vector<UInt> inputDim, columnDim;
     SpatialPooler sp;
+
+    inputDim.push_back(1);
+    columnDim.push_back(1);
+
+    // Must initialize so synPermMax_ gets set to 1.0
+    sp.initialize(inputDim, columnDim);
+
     Real synPermConnected = 0.2;
-    Real synPermActiveInc = 0.05;
     sp.setSynPermConnected(synPermConnected);
-    sp.setSynPermActiveInc(synPermActiveInc);
+
     for (UInt i = 0; i < 100; i++) {
       Real permVal = sp.initPermConnected_();
       NTA_CHECK(permVal >= synPermConnected &&
-                permVal <= synPermConnected + synPermActiveInc / 4.0);
+                permVal <= 1.0);
```

Comment:
Technically this should check that permVal is <= synPermMax_. However that is a protected member so we would need an accessor function. @scottpurdy Any thoughts? (I'm also ok if we actually make synPermMax and synPermMin constants since I can't imagine changing them!)